backend/services/useradm/api/http/api_useradm.go

```diff
@@ -93,359 +108,365 @@ func NewUserAdmApiHandlers(
-	api.Use(
-		// Apply common middlewares
-		&requestlog.RequestLogMiddleware{},
-		&accesslog.AccessLogMiddleware{
-			Format: accesslog.SimpleLogFormat,
-			DisableLog: func(statusCode int, r *rest.Request) bool {
-				if (r.URL.Path == uriInternalAlive ||
-					r.URL.Path == uriInternalHealth) &&
-					statusCode < 300 {
-					return true
-				}
-				return false
-			},
-		},
-		&requestid.RequestIdMiddleware{},
-		&rest.ContentTypeCheckerMiddleware{},
-	)
-	identityMiddleware := &identity.IdentityMiddleware{
-		UpdateLogger: true,
-	}
-	authzMiddleware := &authz.AuthzMiddleware{
-		Authz:              authorizer,
-		ResFunc:            ExtractResourceAction,
-		JWTHandlers:        i.jwth,
-		JWTFallbackHandler: i.config.JWTFallback,
-	}
-	internalRoutes := []*rest.Route{
-		rest.Get(uriInternalAlive, i.AliveHandler),
-		rest.Get(uriInternalHealth, i.HealthHandler),
-
-		rest.Get(uriInternalAuthVerify,
-			authzMiddleware.MiddlewareFunc(
-				identityMiddleware.MiddlewareFunc(i.AuthVerifyHandler),
-			),
-		),
-		rest.Post(uriInternalAuthVerify,
-			authzMiddleware.MiddlewareFunc(
-				identityMiddleware.MiddlewareFunc(i.AuthVerifyHandler),
-			),
-		),
-		rest.Post(uriInternalTenants, i.CreateTenantHandler),
-		rest.Post(uriInternalTenantUsers, i.CreateTenantUserHandler),
-		rest.Delete(uriInternalTenantUser, i.DeleteTenantUserHandler),
-		rest.Get(uriInternalTenantUsers, i.GetTenantUsersHandler),
-		rest.Delete(uriInternalTokens, i.DeleteTokensHandler),
-	}
-	mgmtRoutes := routing.AutogenOptionsRoutes([]*rest.Route{
-		rest.Post(uriManagementAuthLogin, i.AuthLoginHandler),
-		rest.Post(uriManagementAuthLogout, i.AuthLogoutHandler),
-		rest.Post(uriManagementUsers, i.AddUserHandler),
-		rest.Get(uriManagementUsers, i.GetUsersHandler),
-		rest.Get(uriManagementUser, i.GetUserHandler),
-		rest.Put(uriManagementUser, i.UpdateUserHandler),
-		rest.Delete(uriManagementUser, i.DeleteUserHandler),
-		rest.Post(uriManagementSettings, i.SaveSettingsHandler),
-		rest.Get(uriManagementSettings, i.GetSettingsHandler),
-		rest.Post(uriManagementSettingsMe, i.SaveSettingsMeHandler),
-		rest.Get(uriManagementSettingsMe, i.GetSettingsMeHandler),
-		rest.Post(uriManagementTokens, i.IssueTokenHandler),
-		rest.Get(uriManagementTokens, i.GetTokensHandler),
-		rest.Delete(uriManagementToken, i.DeleteTokenHandler),
-		// plans
-		rest.Get(uriManagementPlans, i.GetPlansHandler),
-		rest.Get(uriManagementPlanBinding, i.GetPlanBindingHandler),
-	}, routing.AllowHeaderOptionsGenerator)
-
-	// IdentityMiddleware is only applied to public APIs
-	mgmtRoutes = wrapMiddleware(identityMiddleware, mgmtRoutes...)
-
-	routes := append(mgmtRoutes, internalRoutes...)
-	app, err := rest.MakeRouter(routes...)
+		userAdm:    userAdm,
+		db:         db,
+		jwth:       jwth,
+		config:     config,
+		authorizer: authorizer,
+	}
+}
+
+// Getting false means that a rest error allready been renderd
+func (i *UserAdmApiHandlers) authTokenExtractor(c *gin.Context) (*jwt.Token, bool) {
```

Comment:
The boolean return argument is not really needed as the function returns `nil` whenever it responded. There's also the `gin.Context.ResponseWriter.Written()` that reports whether the header is written.
```suggestion
func (i *UserAdmApiHandlers) authTokenExtractor(c *gin.Context) *jwt.Token {
```

Reply:
I have done that the first time, but then changed it to be safe